src/lib/support/Span.h

```diff
@@ -62,10 +62,19 @@ class Span
     constexpr explicit Span(U (&databuf)[N]) : mDataBuf(databuf), mDataLen(N)
     {}
 
+    // Creates a (potentially mutable) Span view of an std::array
     template <class U, size_t N, typename = std::enable_if_t<sizeof(U) == sizeof(T) && std::is_convertible<U *, T *>::value>>
     constexpr Span(std::array<U, N> & arr) : mDataBuf(arr.data()), mDataLen(N)
```

Comment:
Under what conditions would it be OK to take a non-const lvalue here anyway?  That seems pretty odd...  Is it just that a const lvalue would not return a mutation-allowing pointer from .data()?  That's the thing we should clearly document, I think.

Reply:
Yes std::array actually contains the data, so to convert to a span with writable elements you need a non const array. I would have liked to make that one explicit but there’s code that relies on it being implicit (passing a storage type as a mutable span generally).